app/graph/GraphView.tsx

```diff
@@ -209,7 +209,14 @@ function GraphView({ graph, selectedElement, setSelectedElement, runQuery, histo
 
     const handleRunQuery = async (q: string) => {
         await runQuery(q)
-        chartRef.current?.zoomToFit(1000, 40)
+        // Get canvas dimensions
```

Comment:
Can you move that to a function and call this function from here and handleCenterClick ?